src/AWS.Deploy.Common/Recipes/OptionSettingItem.cs

```diff
@@ -38,9 +38,10 @@ public interface IOptionSettingItem
         /// <summary>
         /// Set the value of an <see cref="OptionSettingItem"/> while validating the provided input.
         /// </summary>
-        /// <param name="optionSettingHandler"></param>
-        /// <param name="value"></param>
-        void SetValue(IOptionSettingHandler optionSettingHandler, object value);
+        /// <param name="optionSettingHandler">Handler use to set any child option settings</param>
+        /// <param name="validatorFactory">Source of validators for this item</param>
+        /// <param name="value">Value to set</param>
+        void SetValue(IOptionSettingHandler optionSettingHandler, IValidatorFactory validatorFactory, object value);
```

Comment:
could you remove the `IValidatorFactory` as a dependency to `SetValue` and instead pass in the list of validators, which is the result of `validatorFactory.BuildValidators(this)`? `SetOptionSettingValue` in `OptionSettingHandler` is meant to handle anything related to services and only pass down simplified objects to `SetValue` in `OptionSettingItem`.

Reply:
Done